Library/Homebrew/cask/lib/hbc/cli/search.rb

```diff
@@ -2,8 +2,12 @@ module Hbc
   class CLI
     class Search < AbstractCommand
       def run
-        results = self.class.search(*args)
-        self.class.render_results(*results)
+        if args.empty?
+          puts Formatter.columns(CLI.nice_listing(Hbc.all_tokens))
```

Comment:
This should only be called if there is no connection and should probably be handled in `::search`.

Reply:
The last thing here would be to update the man page to mention this, other than that it looks good to me. (Make sure to run `brew man` afterwards.)